### ISSUE TITLE
Cannot call fit() twice in EBM binary classification mode

### ISSUE THREAD

**Issue by kevencarneiro** [interpretml/interpret#312]
When trying to do binary classification with EBM I'm receiving the error below if I fit a dataset twice.

I'm using interpret==0.2.7 from pip.

## Explanation
```python
ebm = ExplainableBoostingClassifier()
ebm.fit(X_train, y_train) # works
ebm.fit(X_train, y_train) # ValueError: Shape of passed values is (112, 4), indices imply (112, 10)
```

Note that this is failing only for binary classification. If I remove the `y = y == 'setosa'` line, everything works as expected.

## Complete code (with dataset load)
```python
from interpret.glassbox import ExplainableBoostingClassifier
from sklearn.model_selection import train_test_split
import pandas as pd

file_name = "https://raw.githubusercontent.com/uiuc-cse/data-fa14/gh-pages/data/iris.csv"
df = pd.read_csv(file_name)
df.head()

train_cols = df.columns[0:-1]
label = df.columns[-1]
X = df[train_cols]
y = df[label].astype('category')
y = y == 'setosa'
X_train, X_test, y_train, y_test = train_test_split(X, y)

ebm = ExplainableBoostingClassifier()
ebm.fit(X_train, y_train)
ebm.fit(X_train, y_train)
```

## Full traceback
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/usr/local/lib/python3.7/dist-packages/pandas/core/internals/managers.py in create_block_manager_from_blocks(blocks, axes)
   1670                 blocks = [
-> 1671                     make_block(values=blocks[0], placement=slice(0, len(axes[0])))
   1672                 ]

8 frames
/usr/local/lib/python3.7/dist-packages/pandas/core/internals/blocks.py in make_block(values, placement, klass, ndim, dtype)
   2743 
-> 2744     return klass(values, ndim=ndim, placement=placement)
   2745 

/usr/local/lib/python3.7/dist-packages/pandas/core/internals/blocks.py in __init__(self, values, placement, ndim)
    130             raise ValueError(
--> 131                 f"Wrong number of items passed {len(self.values)}, "
    132                 f"placement implies {len(self.mgr_locs)}"

ValueError: Wrong number of items passed 4, placement implies 10

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-25-c70460f598ca> in <module>()
----> 1 ebm.fit(X_train, y_train)

/usr/local/lib/python3.7/dist-packages/interpret/glassbox/ebm/ebm.py in fit(self, X, y, sample_weight)
    884             privacy_schema=getattr(self, 'privacy_schema', None)
    885         )
--> 886         self.preprocessor_.fit(X)
    887         X_orig = X
    888         X = self.preprocessor_.transform(X_orig)

/usr/local/lib/python3.7/dist-packages/interpret/glassbox/ebm/ebm.py in fit(self, X)
    182         native = Native.get_native_singleton()
    183         schema = autogen_schema(
--> 184             X, feature_names=self.feature_names, feature_types=self.feature_types
    185         )
    186 

/usr/local/lib/python3.7/dist-packages/interpret/utils/all.py in autogen_schema(X, ordinal_max_items, feature_names, feature_types)
    372         # As used from SO:
    373         # https://stackoverflow.com/questions/47393134/attributeerror-dataframe-object-has-no-attribute-infer-objects
--> 374         X = pd.DataFrame(X, columns=feature_names)
    375         try:
    376             X = X.infer_objects()

/usr/local/lib/python3.7/dist-packages/pandas/core/frame.py in __init__(self, data, index, columns, dtype, copy)
    495                 mgr = init_dict({data.name: data}, index, columns, dtype=dtype)
    496             else:
--> 497                 mgr = init_ndarray(data, index, columns, dtype=dtype, copy=copy)
    498 
    499         # For data is list-like, or Iterable (will consume into list)

/usr/local/lib/python3.7/dist-packages/pandas/core/internals/construction.py in init_ndarray(values, index, columns, dtype, copy)
    232         block_values = [values]
    233 
--> 234     return create_block_manager_from_blocks(block_values, [columns, index])
    235 
    236 

/usr/local/lib/python3.7/dist-packages/pandas/core/internals/managers.py in create_block_manager_from_blocks(blocks, axes)
   1679         blocks = [getattr(b, "values", b) for b in blocks]
   1680         tot_items = sum(b.shape[0] for b in blocks)
-> 1681         raise construction_error(tot_items, blocks[0].shape[1:], axes, e)
   1682 
   1683 

ValueError: Shape of passed values is (112, 4), indices imply (112, 10)
```
## Working example on Google Colab
[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1uwa6dFtNgwBlswqYEzxXwM1IUDkLjNmz?usp=sharing)

**Comment by interpret-ml** [comment]
Hi @kevencarneiro, 

Thanks for reporting this! This should be getting fixed in our next release, and relates to an incompatibility we previously had with the scikit-learn API. For the time being, we recommend re-initializing the model object by calling the constructor again -- it'll take an extra line of code but should produce the same results you expect.

-InterpretML Team

**Comment by paulbkoch** [comment]
This should be fixed in our latest v0.3.0 release.  Please re-open if you're still seeing the same failure.
